### DISCUSSION TITLE
Retain chapters when merging two videos

### DISCUSSION THREAD

**Discussion by jadoff630** [mifi/lossless-cut#2164] [Q&A]
I have two videos that I am trying to merge into one. The merging goes well, but I am trying to keep the chapter markers from both files when I combine. Right now when I combine, only the chapters from the first file are there. 

I opened each file in the editor viewer before, selecting to import chapters, and there are now proj.llc files for the two videos. 

In merge settings, I have checked the following options: 
- "Include all tracks...." 
- "preserve original metadata..." 
- "Preserve all MP4/MOV metadata"

**Comment by mifi** [comment]
Unfortunately I don't know how to easily fix this. Seems to be an open issue in ffmpeg: https://trac.ffmpeg.org/ticket/6468

**Reply by NeoPlayer13** [reply]
Here is a solution I found using the original ffmpeg ticket as reference. [Link](https://gist.github.com/FabianBartl/16396586d5022b422817c6c3d5ec4d26)

***Updated link**

**Reply by DashingDave** [reply]
Is there a new link for this? It now leads to a 404 error.

**Reply by NeoPlayer13** [reply]
Try now @DashingDave it should be working.

**Reply by DashingDave** [reply]
Yes, the link works now--thanks!

Now, after opening it in Mac's Python Launcher, which then entered some code into a Terminal window that ends with

`Usage: /Users/dashingdave/Downloads/PythonCode/merge-videos.py [input file] [input file] [output file]`

 (where PythonCode is the name of the folder container the .py files) which I interpreted to mean enter 

`/Users/dashingdave/Downloads/PythonCode/merge-videos.py /Users/dashingdave/Downloads/PythonCode/Clip1.m4v /Users/dashingdave/Downloads/PythonCode/Clip2.m4v /Users/dashingdave/Downloads/PythonCode/CombinedClip.m4v`

into a second Terminal window in order to combine Clip1 and Clip2 (both placed into the same folder as the .py files) into CombinedClip.

However, this gives me a permissions error:

`zsh: permission denied: /Users/dashingdave/Downloads/PythonCode/merge-videos.py`

What am I doing incorrectly?

**Reply by NeoPlayer13** [reply]
I'm not really a Mac user, so I don't have any experience with Mac's Python Launcher, but if it follows the same logic as a normal Linux OS. That error should referring to it missing the execute permission.

**Comment by Dragodraki** [comment]
> I have two videos that I am trying to merge into one. The merging goes well, but I am trying to keep the chapter markers from both files when I combine. Right now when I combine, only the chapters from the first file are there.

I have this problem as well, its frustrating...

I am not very familiar with Python or ffmpeg command line at all. Could you please fix this in your next version?
